lxd/images.go

```diff
@@ -806,7 +806,16 @@ func imagesPost(d *Daemon, r *http.Request) response.Response {
 		return nil
 	}
 
-	op, err := operations.OperationCreate(d.State(), project, operations.OperationClassTask, db.OperationImageDownload, nil, nil, run, nil, nil)
+	var metadata interface{}
+
+	secret, _ := shared.RandomCryptoString()
+	if secret != "" {
+		metadata = map[string]string{
+			"secret": secret,
+		}
+	}
```

Comment:
Shouldn't this be limited to only when the source data says we're pushing?

Reply:
In the case of a push, `req` won't be of type `api.ImagesPost{}`. But we can check whether metadata has been provided as that's only done when pushing.